### PR TITLE
Feat/add tracing id to log and errors

### PR DESCRIPTION
**Pull request by marcofenske-vi**
Add Spring Sleuth to have tracing Ids in log messages and support distributed tracing.
Also use traceIds as ids for error response objects.

### CODE REVIEW

File: src/main/java/com/myCompanyName/myProjectName/common/rest/error/RestExceptionHandler.java [hunk 1]
```diff
@@ -14,8 +14,13 @@
 @ControllerAdvice
 public class RestExceptionHandler {
 
-  private RestErrorBuilder restErrorBuilder = new RestErrorBuilder();
-  private RestErrorLogger restErrorLogger = new RestErrorLogger();
+  private RestErrorBuilder restErrorBuilder;
+  private RestErrorLogger restErrorLogger;
```
Comment (RafaelMarquesVI) [inline src/main/java/com/myCompanyName/myProjectName/common/rest/error/RestExceptionHandler.java:18]: these could be final

File: src/main/java/com/myCompanyName/myProjectName/common/rest/error/RestExceptionHandler.java [hunk 2]
```diff
@@ -14,8 +14,13 @@
...
+
+  public RestExceptionHandler(final RestErrorBuilder restErrorBuilder, final RestErrorLogger restErrorLogger) {
```
Comment (RafaelMarquesVI) [inline src/main/java/com/myCompanyName/myProjectName/common/rest/error/RestExceptionHandler.java:20]: aren't these autowired?

File: src/main/java/com/myCompanyName/myProjectName/common/rest/error/RestErrorBuilder.java [hunk 3]
```diff
@@ -1,15 +1,35 @@
...
+@Component
 public class RestErrorBuilder {
 
-  public Error buildError(Exception e, RestErrorCode errorCode) {
-    return new Error()
-        .id(UUID.randomUUID().toString())
-        .code(errorCode.getErrorCode())
-        .message(e.getMessage());
-  }
+    private Tracer tracer;
```
Comment (RafaelMarquesVI) [inline src/main/java/com/myCompanyName/myProjectName/common/rest/error/RestErrorBuilder.java:16]: could this be final?

File: src/main/java/com/myCompanyName/myProjectName/common/rest/error/RestErrorBuilder.java [hunk 4]
```diff
@@ -1,15 +1,35 @@
...
+
+    public RestErrorBuilder(final Tracer tracer) {
```
Comment (RafaelMarquesVI) [inline src/main/java/com/myCompanyName/myProjectName/common/rest/error/RestErrorBuilder.java:18]: should this be autowired?

**Comment by frittinger** [comment]
I like this addition!
I saw that there are plugins for queues which could be very interesting for IFX for the purging. However, I was wondering if there are also mechanisms to trace from the frontend (Javascript) through the backend. So to have a trace id in the log for a distinct AJAX call or similar from the frontend? I found nothing in the docs.

File: src/main/java/com/myCompanyName/myProjectName/common/rest/error/RestExceptionHandler.java (hunk 2 above)
Comment (marcofenske-vi) [reply src/main/java/com/myCompanyName/myProjectName/common/rest/error/RestExceptionHandler.java:20]: this IS autowiring (by constructor) 
Field injection is a bad practice, see https://reflectoring.io/constructor-injection/

File: src/main/java/com/myCompanyName/myProjectName/common/rest/error/RestErrorBuilder.java (hunk 4 above)
Comment (marcofenske-vi) [reply src/main/java/com/myCompanyName/myProjectName/common/rest/error/RestErrorBuilder.java:18]: constructor injection (se above)

File: README.md [hunk 5]
```diff
@@ -81,6 +81,9 @@ The module model is explained [here](https://github.com/odrotbohm/moduliths#modu
 
 In the end it's about reducing accessibility of classes to a minimum, hiding internals of modules and preventing cyclic dependencies.  
 
+## Logging
+[Spring Sleuth](https://spring.io/projects/spring-cloud-sleuth) is included in this boilerplate to support trace ids which are added to all log messages that originate from a request. 
+Further, trace ids are also forwarded automatically to other services called by Spring's RestTemplate that are also based on this boilerplate or include Spring Sleuth. This means that distributed tracing is supported out of the box.
```
Comment (ImpLarsBruemmer) [inline README.md:86]: Hi @marcofenske-vi,

could you please provide an example log message to see which fields are actually available and how the format does look like?

**Review by ImpLarsBruemmer — CHANGES_REQUESTED** [review]

File: README.md (hunk 5 above)
Comment (marcofenske-vi) [reply README.md]: Done

**Review by ImpLarsBruemmer — APPROVED** [review]

**Pull request merged**